File: java/oop/InterfaceManApp/src/com/techlabs/imanable/test/TestIManAble.java
package com.techlabs.imanable.test;

import com.techlabs.imanable.Boy;
import com.techlabs.imanable.IEmotionAble;
import com.techlabs.imanable.IMannerAble;
import com.techlabs.imanable.Man;

public class TestIManAble {
	public static void main(String[] args) {
		Man man=new Man();
		Boy boy=new Boy();
		
		atTheParty(man);
		atTheParty(boy);
		//atTheCinemas(man);
		atTheCinemas(boy);
	}
	
	public static void atTheParty(IMannerAble iMannerAble){
		iMannerAble.wish();
		iMannerAble.depart();
	}
	
	public static void atTheCinemas(IEmotionAble iEmotionAble){
		iEmotionAble.cry();
		iEmotionAble.laugh();
	}
}

File: dotNet/CSharp/InterfaceManApp/InterfaceManApp/Test/TestIManAble.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace InterfaceManApp.Test
{
    public class TestIManAble
    {
        static void Main(string[] args)
        {
            Man man = new Man();
            Boy boy = new Boy();

            AtTheParty(man);
            AtTheParty(boy);

            //AtTheCinemas(man);
            AtTheCinemas(boy);
        }

        public static void AtTheParty(IMannerAble iMannerAble)
        {
            iMannerAble.Wish();
            iMannerAble.Depart();
        }

        public static void AtTheCinemas(IEmotionAble iEmotionAble)
        {
            iEmotionAble.Cry();
            iEmotionAble.Laugh();
        }
    }
}
